### PR TITLE
Few more negative tests for mnemonic rewards-credentials

### PR DESCRIPTION
**Pull request by piotr-iohk**
# Issue Number

#1119

# Overview

- [ ] Few more negative tests for mnemonic rewards-credentials

# Comments

Two are failing, don't know why...

### CODE REVIEW

File: lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Mnemonics.hs [hunk 1]
```diff
@@ -47,14 +56,87 @@ spec :: forall t. KnownCommand t => SpecWith ()
...
+        T.unpack out `shouldContain` mempty
+
+    it "CLI_MNEMONICS_REWARD_CREDENTIALS - \
+        \cannot use 2nd mnemonic > 12" $ do
+        Stdout m1 <- generateMnemonicsViaCLI @t []
+        Stdout m2 <- generateMnemonicsViaCLI @t ["--size", "18"]
+        (c, out, err)  <- getRewardCredentialsViaCli @t (T.pack m1) (T.pack m2)
+        c `shouldBe` ExitFailure 1
+        T.unpack err `shouldContain` "Invalid number of words: 9 or 12 words are expected."
+        T.unpack out `shouldContain` mempty
```
Comment (piotr-iohk) [inline lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Mnemonics.hs]: I don't know why those cases fail. They seem to work fine _manually_...
`cannot use 2nd mnemonic < 9`
`cannot use 2nd mnemonic > 12`

Comment (KtorZ) [reply lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Mnemonics.hs]: @piotr-iohk what do you mean? What does fail exactly?

Comment (KtorZ) [reply lib/jormungandr/test/integration/Test/Integration/Jormungandr/Scenario/CLI/Mnemonics.hs]: _ok I see now when running the tests_

**Review by KtorZ — APPROVED** [review]
:+1: reviewed and also extended the "happy" scenario with an exhaustive test matrix coverage all possibilities:

```
    CLI_MNEMONICS_REWARD_CREDENTIALS - address obtained with credentials matches JCLI
      --size=default (seed), --size=default (snd factor)
      --size=default (seed), --size=9 (snd factor)
      --size=default (seed), --size=12 (snd factor)
      --size=15 (seed), --size=default (snd factor)
      --size=15 (seed), --size=9 (snd factor)
      --size=15 (seed), --size=12 (snd factor)
      --size=18 (seed), --size=default (snd factor)
      --size=18 (seed), --size=9 (snd factor)
      --size=18 (seed), --size=12 (snd factor)
      --size=21 (seed), --size=default (snd factor)
      --size=21 (seed), --size=9 (snd factor)
      --size=21 (seed), --size=12 (snd factor)
      --size=24 (seed), --size=default (snd factor)
      --size=24 (seed), --size=9 (snd factor)
      --size=24 (seed), --size=12 (snd factor)
```

**Pull request merged**